src/Workspaces/Core/Portable/Rename/RenameLocation.ReferenceProcessing.cs

```diff
@@ -337,12 +344,20 @@ private static string TrimNameToAfterLastDot(string name)
                             {
                                 if (location.IsInSource)
                                 {
-                                    var token = location.FindToken(cancellationToken);
-                                    if (!syntaxFacts.IsReservedOrContextualKeyword(token) &&
-                                        token.ValueText == referencedSymbol.Name)
+                                    var document = solution.GetRequiredDocument(location.SourceTree);
+
+                                    // If the location is in a source generated file, we won't rename it. Our assumption in this case is we
+                                    // have cascaded to this symbol from our original source symbol, and the generator will update this file
+                                    // based on the renamed symbol.
+                                    if (document is not SourceGeneratedDocument)
                                     {
-                                        Contract.ThrowIfNull(location.SourceTree);
-                                        results.Add(new RenameLocation(location, solution.GetRequiredDocument(location.SourceTree).Id));
+                                        var token = location.FindToken(cancellationToken);
+                                        if (!syntaxFacts.IsReservedOrContextualKeyword(token) &&
+                                            token.ValueText == referencedSymbol.Name)
+                                        {
+                                            Contract.ThrowIfNull(location.SourceTree);
```

Comment:
This must be not null if you were able to get the document from it, correct?

Reply:
Oh right; the only reason it was there was before we annotated the compiler to tell it that IsInSource means SourceTree is non-null. But that's annotated now. Will delete.